### PR TITLE
community: AWS Athena Document Loader

### PR DESCRIPTION
**Pull request by abhijeethp**
- **Description:** Adds the document loader for [AWS Athena](https://aws.amazon.com/athena/), a serverless and interactive analytics service.
  - **Dependencies:** Added boto3 as a dependency

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
please do not update langchain, should be imported from langchain_community from now on

other than that, lgtm!

**Comment by abhijeethp** [comment]
@hwchase17 I have removed the extraneous imports from langchain... but I believe the rest should stay unless you are saying that the end users would no longer be importing this as `from langchain.document_loaders import AthenaLoader` and rather would be importing as `from langchain_community.document_loaders.athena import AthenaLoader`

please confirm and I will remove the rest of the changes under langchain as well.

File: libs/community/langchain_community/document_loaders/athena.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+import io
+import json
+import time
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+
+import pandas as pd
```
Comment (baskaryan) [inline libs/community/langchain_community/document_loaders/athena.py:8]: pandas isn't a required dependency so it shouldn't be imported at the top level. should instead import in the functions that use it

Comment (abhijeethp) [reply libs/community/langchain_community/document_loaders/athena.py]: Thanks for the feedback. 
I Have made this change

**Comment by abhijeethp** [comment]
@hwchase17 @baskaryan please review

**Review by baskaryan — COMMENTED** [review]
cc @3coins

**Review by 3coins — COMMENTED** [review]
@abhijeethp 
Thanks for working on these changes. Left some comments about the athena and s3 clients, and some suggestions on re-organizing the code. You also seem to have minor failures in the attached notebook, can you run the linter and fix these errors. Let me know if you need any help with linting or fixing any of these errors.

**Comment by abhijeethp** [comment]
@3coins Thanks for the feedback! I have made the suggested changes. 
I am unfortunately not able to fix the linting issues in the python notebooks. 
I do not see a way to run linter in the docs directory... Please let me know how i can fix them.

**Comment by baskaryan** [comment]
> @3coins Thanks for the feedback! I have made the suggested changes. I am unfortunately not able to fix the linting issues in the python notebooks. I do not see a way to run linter in the docs directory... Please let me know how i can fix them.

running `make format; make lint` from root of repo should run docs formatting and linting! happy to help if you want to give me write access to your fork

**Comment by abhijeethp** [comment]
@baskaryan Thanks for the feedback. 
I was able to fix the formatting and linting issues by running it from root.

**Comment by abhijeethp** [comment]
@hwchase17 why did was the PR closed?

**Comment by baskaryan** [comment]
not intentional, we can't reopen at the moment but will do so as soon as we can :/ See #16796

**Comment by abhijeethp** [comment]
@3coins I have made the changes suggested above. 
Please review it again when possible.
Would love to have this be part of the library soon! :D

**Review by hwchase17 — APPROVED** [review]

**Comment by abhijeethp** [comment]
@hwchase17 I had to make a change to get it working for python 3.8 
https://github.com/langchain-ai/langchain/pull/15625/commits/1428649aaa330840aa15697a5211f445159f831e

These builds only ran on approval so i could not catch it earlier. 
Please review it once again

**Pull request merged**